multiaddr.go

```diff
@@ -113,3 +113,32 @@ func (m *multiaddr) Decapsulate(o Multiaddr) Multiaddr {
 	}
 	return ma
 }
+
+var ErrProtocolNotFound = fmt.Errorf("protocol not found in multiaddr")
+
+func (m *multiaddr) ValueForProtocol(code int) (string, error) {
```

Comment:
This could use `Split(m)`. Or maybe have a version that yields `[]struct{int, string}` -- maybe: `ProtocolValues(m) -- and this could just be:

```go
func (m *multiaddr) ValueForProtocol(code int) (string, error) {
  for _, part := range m.ProtocolValues() {
    if part.Code == code {
      return part.Value
    }
  }
  return nil, ErrProtocolNotFound
}
```

(This method is calling `m.String()` which is an expensive operation anyway: https://github.com/jbenet/go-multiaddr/blob/master/multiaddr.go#L47-L53 )